practice.rb

```diff
@@ -0,0 +1,50 @@
+## Problem #1
+def print_array(array_of_integers)
+  print array_of_integers
+end
+
+## Problem #2
+def stretch(array)
+  stretch = []
+  array.each do |i|
+    if i % 2 == 0
+      2.times do
+        stretch << i / 2
+      end
+    else
+      stretch << (i / 2) + 1
+      stretch << (i / 2)
+    end
+  end
+  return stretch
+end
+
+## Problem #3
+def numUnique(integer_array)
+  numbers = {}
+  integer_array.each do |input|
+    if !numbers.has_key?(input)
+      numbers[input] = 1
+    end
+  end
+  return numbers.length
+end
```

Comment:
Like this?: 
def numUnique(integer_array)
  numbers = {}
  integer_array.each do |input|
    if !numbers.has_key?(input)
      numbers[input] = 1
    else
      numbers[input] += 1
    end
  end
  return numbers.length
end